raft/raft.go

```diff
@@ -565,6 +565,12 @@ func (r *raft) Step(m pb.Message) error {
 	case m.Term > r.Term:
 		lead := m.From
 		if m.Type == pb.MsgVote {
+			if r.state == StateFollower && r.checkQuorum && r.electionElapsed < r.electionTimeout {
```

Comment:
Why is this limited to followers? As long as `checkQuorum` is enabled, any node in `StateLeader` is guaranteed to have been in contact with a majority of nodes within an election timeout, so it shouldn't increase its term either. 

> If the entire cluster is restarted or is bootstrapped for the first time, there will be a few seconds unavailability.

This is already true: when the entire cluster is restarted, the first election cannot happen until an election timeout (plus randomization) has passed.